src/java_bytecode/expr2java.cpp

```diff
@@ -237,23 +237,40 @@ std::string expr2javat::convert_constant(
     dest+='L';
     return dest;
   }
-  else if (src.type() == java_float_type()) {
-    ieee_floatt Fpn(to_constant_expr(src));
-    float F = Fpn.to_float();
-    std::stringstream buffer;
-    buffer << std::hexfloat;
-    buffer << F << 'f';
-    return(buffer.str());
-  }
-   else if (src.type() == java_double_type()) {
-    ieee_floatt Fpn(to_constant_expr(src));
-    double D = Fpn.to_double();
+  else if((src.type()==java_float_type()) ||
+          (src.type()==java_double_type()))
+  {
+    ieee_floatt ieee_repr(to_constant_expr(src));
+    if(ieee_repr.is_NaN() || ieee_repr.is_infinity())
+    {
+      std::string result;
+      if(ieee_repr.get_sign())
+        result="-";
+      else
+        result="+";
+      if(ieee_repr.is_NaN())
+        result+="NaN";
+      else
+        result+="inf";
```

Comment:
for Java, this is `Double/Float.NEGATIVE_INFINITY` or `Double/Float.POSITIVE_INFINITY`
(cf. https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html)